Strata/DDM/AST.lean

```diff
@@ -940,9 +1022,19 @@ instance (nm : String) (d : Dialect) : Decidable (nm ∈ d) :=
 
 end Dialect
 
+
+/-- BEq between two Std HashMap; checked by doing inclusion test twice -/
+instance [BEq α] [Hashable α] [BEq β]: BEq (Std.HashMap α β) where
+  beq x y :=
+    let includes (x y:Std.HashMap α β) :=
+      let l := x.toList
+      l.all (fun (k,v) => y.get? k == .some v)
+    includes x y && includes y x
```

Comment:
I'm not complete sure on timing, but this could be optimized to one loop and avoid incremental list construction using the pigeon hole principal and do syntax:
```
  beq x y := Id.run do
    if x.size ≠ y.size then
      return false
    for (k, v) in x do
      if y.get? k != Option.some v then
        return false
    return true
```